iib/workers/tasks/opm_operations.py

```diff
@@ -295,196 +295,132 @@ def inner(*args, **kwargs):
     return decorator
 
 
-def opm_serve_from_index(base_dir: str, from_index: str) -> Tuple[int, subprocess.Popen]:
+def get_list_packages(
+    input_data: str,
+    base_dir: str,
+) -> List[str]:
     """
-    Locally start OPM registry service, which can be communicated with using gRPC queries.
-
-    Due to IIB's paralellism, the service can run multiple times, which could lead to port
-    binding conflicts. Resolution of port conflicts is handled in this function as well.
+    Run OPM render and get list of olm.package names.
 
-    :param str base_dir: base directory to create temporary files in.
-    :param str from_index: index image to inspect.
-    :return: tuple containing port number of the running service and the running Popen object.
-    :rtype: (int, Popen)
+    :param str input_data: input data for opm render
+        Example: catalog-image | catalog-directory | bundle-image | bundle-directory | sqlite-file
+    :param str base_dir: temp directory where opm will be executed.
+    :return: list of package names present in input data.
+    :rtype: [str]
     """
-    from iib.workers.tasks.build import _get_index_database
-
-    log.info('Serving data from image %s', from_index)
-    if not is_image_fbc(from_index):
-        db_path = _get_index_database(from_index, base_dir)
-        return opm_registry_serve(db_path=db_path)
+    from iib.workers.tasks.utils import run_cmd
 
-    catalog_dir = get_catalog_dir(from_index, base_dir)
-    return opm_serve(catalog_dir=catalog_dir)
+    input_data_path = _get_input_data_path(input_data, base_dir)
 
+    cmd = [Opm.opm_version, 'render', input_data_path]
 
-@create_port_filelocks(port_purposes=["opm_port", "opm_pprof_port"])
-def opm_serve(
-    opm_port: int,
-    catalog_dir: str,
-    opm_pprof_port: Optional[int] = None,
-) -> Tuple[int, subprocess.Popen]:
-    """
-    Locally start OPM service, which can be communicated with using gRPC queries.
+    opm_render_output = run_cmd(
+        cmd, {'cwd': base_dir}, exc_msg=f'Failed to list packages on {input_data}'
+    )
 
-    Due to IIB's paralellism, the service can run multiple times, which could lead to port
-    binding conflicts. Resolution of port conflicts is handled in this function as well.
+    if not opm_render_output:
+        log.info("There are no packages in %s", input_data)
+        return []
 
-    :param int opm_port: OPM port number obtained from create_port_filelock decorator
-    :param int opm_pprof_port: Pprof opm port number obtained from create_port_filelock decorator
-    :param str catalog_dir: path to file-based catalog directory that should be served.
-    :return: tuple containing port number of the running service and the running Popen object.
-    :rtype: (int, Popen)
-    """
-    log.info('Serving data from file-based catalog %s', catalog_dir)
+    olm_packages = [
+        json.loads(package) for package in re.split(r'(?<=})\n(?={)', opm_render_output)
```

Comment:
Just out of curiosity: Any specific reason we're splitting the content between brackets and then loading into `json`? Couldn't we simply load the JSON list and parse its dictionaries?

Reply:
The parser does not work that way. Since the opm output is not valid JSON. We get something like this:
```
{
...
data
...
}
{
....
data2
...
}
```
The whole output is not a list with multiple dictionaries. Those are just dictionaries so JSON package can't parse it correctly. And the easiest way to do it is to split it.

Reply:
Got it, thanks!